Too many database connections are being opened

##### Brief description of problem/feature
When I try to connect to our prod databases via the command line, I'm getting the following error:
```
psql: error: FATAL:  remaining connection slots are reserved for non-replication superuser connections
```

This had happened about a year ago as well to a worse degree, where we were getting proxy errors on our servers. It's possible that this has been happening occasionally since then though.

Unfortunately, I did not provide enough detail in my email thread with @mechanicjay... I even say that I thought I figured out the offending code, but I didn't say where that was, and I don't see any commits in the that timeframe that would have addressed it :woozy_face:  What we did conclude is that our application is creating too many database connections for some reason.

From me in the thread:
>Based on this and some more testing I just did, I think I've found the offending code. Not sure why we are creating quite that many connections right now, but I'll see what I can do.

And later on this is me quoting @mechanicjay and responding:
>>Based on observed behavior right now, it looks sidewalk opens 4 connections to the db when it starts and sits on them, the appears to open/close connections as needed while the software does things
>
>I'm seeing something similar. I'm also seeing that after the code that requires more connections is run, our application might just not be closing those connections in a timely fashion? I ran the code that brought the sidewalk-pittsburgh database to jump up to 26 connections, but the code has finished running, and all of those connections are still open.
>
>Do you think the issue is that each application is not letting go of connections, and since we are actually running 10 copies of the application, the connection pooler isn't aware of the other applications and doesn't know to stop hoarding the connections? 

##### Potential solution(s)
We'll want to test our application on our local dev env, looking at how many connections are being opened. Then we should try doing different operations to figure out what might be causing the spike in database connections. I think I remember it being queries for Validate and/or Gallery that were the culprit.

Here's the exact error message from the Auckland server logs:
>2023-07-27 22:42:45,583 - [ERROR] - from com.jolbox.bonecp.hooks.AbstractConnectionHook in play-internal-execution-context-1
>Failed to acquire connection to jdbc:postgresql://docker-test.cs.washington.edu:5433/sidewalk-auckland Sleeping for 1000ms and trying again. Attempts left: 10. Exception: null.Message:FATAL: remaining connection slots are reserved for non-replication superuser connections

--- a/conf/application.conf
+++ b/conf/application.conf
@@ -39,6 +39,9 @@ db.default.url="jdbc:postgresql://localhost:5432/sidewalk"
 # db.default.user=sa
 db.default.user="sidewalk"
 db.default.password="sidewalk"
+db.default.idleMaxAge=30 second  # default is 10 minutes
+db.default.acquireRetryDelay=3 second  # default is 1 second
+db.default.acquireRetryAttempts=20  # default is 10
 
 
 #override default url if environment variable is present

--- a/conf/application.conf
+++ b/conf/application.conf
@@ -39,9 +39,10 @@ db.default.url="jdbc:postgresql://localhost:5432/sidewalk"
 # db.default.user=sa
 db.default.user="sidewalk"
 db.default.password="sidewalk"
-db.default.idleMaxAge=30 second  # default is 10 minutes
-db.default.acquireRetryDelay=3 second  # default is 1 second
-db.default.acquireRetryAttempts=20  # default is 10
+db.default.idleMaxAge=2 minute  # default is 10 minutes
+db.default.acquireRetryDelay=5 second  # default is 1 second
+db.default.acquireRetryAttempts=42  # default is 10
+db.default.minConnectionsPerPartition=4  # default is 5
 
 
 #override default url if environment variable is present

